oak_restricted_kernel_sdk_proc_macro/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Item, Result};
+
+#[proc_macro_attribute]
+pub fn entrypoint(_attr: TokenStream, input: TokenStream) -> TokenStream {
+    let input_item = parse_macro_input!(input as syn::Item);
+
+    match process_oak_main(input_item) {
+        Ok(output) => output,
+        Err(error) => error.to_compile_error().into(),
+    }
+}
+
+fn process_oak_main(input_item: Item) -> Result<TokenStream> {
+    let input_fn = match input_item {
+        syn::Item::Fn(fn_item) => Ok(fn_item),
```

Comment:
Could you check the type of the function too? Otherwise I think the errors may not be very obvious.

Reply:
Added checks for the type signature. Will add compile tests in a separate pr.

Reply:
adding tests turned out to be a bit more complicated than anticipated. There's https://github.com/Manishearth/compiletest-rs, but uses the raw rustc and doesn't play nice with dependencies. Probs fine to just make sure it works with our examples.

Reply:
yeah I think compile tests are not necessary, but thanks for looking into it